lib/auth/password.go

```diff
@@ -166,7 +166,7 @@ func (a *Server) ChangePassword(ctx context.Context, req *proto.ChangePasswordRe
 func (a *Server) checkPasswordWOToken(user string, password []byte) error {
 	const errMsg = "invalid username or password"
 
-	err := services.VerifyPassword(password)
+	err := services.VerifyPassword(password, false)
```

Comment:
Is verifying existing passwords even necessary? Could we only validate the rules on creation?

Otherwise +1 for using separate functions, as Tim suggested - https://github.com/gravitational/teleport/pull/36579#pullrequestreview-1816444652.

Reply:
⬆️ This.

Reply:
I kept validation for maximum password length, thinking it would be helpful against some potential DoS, but perhaps it's not needed. Let me know your thoughts.